dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -2748,7 +2801,12 @@ RtpsUdpDataLink::RtpsWriter::process_acknack(const RTPS::AckNackSubmessage& ackn
 
     if (reader->expecting_ack_) {
       reader->expecting_ack_ = false;
-      link->received_expected_ack();
+      --expected_acks_;
+      last_ack_ = MonotonicTimePoint::now();
+      if (expected_acks_ == 0) {
+        heartbeat_.cancel();
+        heartbeat_.schedule(TimeDuration::zero_value);
```

Comment:
Ditto cancel estimate question here, maybe even more so since the deadline is zero.

Reply:
The goal here was to get to processing the nack replies as soon as possible, i.e., when a nack from every reader has been received.  Based on the wait for ack test on my machine, the overhead is .000014 seconds.  This will only occur once per heartbeat period (say 100ms).